packages/dashboard-frontend/src/pages/GetStarted/ImportFromGit/index.tsx

```diff
@@ -18,39 +18,67 @@ import {
   Form,
   FormGroup,
   FormHelperText,
-  FormSection,
+  Panel,
+  PanelHeader,
+  PanelMain,
+  PanelMainBody,
   TextInput,
+  Title,
   ValidatedOptions,
 } from '@patternfly/react-core';
 import { ExclamationCircleIcon } from '@patternfly/react-icons';
 import { History } from 'history';
 import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
 
 import { FactoryLocationAdapter } from '@/services/factory-location-adapter';
+import { EDITOR_ATTR, EDITOR_IMAGE_ATTR } from '@/services/helpers/factoryFlow/buildFactoryParams';
 import { buildUserPreferencesLocation } from '@/services/helpers/location';
 import { UserPreferencesTab } from '@/services/helpers/types';
+import { AppState } from '@/store';
+import { selectSshKeys } from '@/store/SshKeys/selectors';
+import * as WorkspacesStore from '@/store/Workspaces';
 
-export type Props = {
-  hasSshKeys: boolean;
+export type Props = MappedProps & {
   history: History;
+  editorId: string;
+  editorImage: string | undefined;
 };
 export type State = {
+  hasSshKeys: boolean;
   location: string;
   validated: ValidatedOptions;
 };
 
-export class ImportFromGit extends React.PureComponent<Props, State> {
+class ImportFromGit extends React.PureComponent<Props, State> {
   constructor(props: Props) {
     super(props);
 
     this.state = {
+      hasSshKeys: this.props.sshKeys.length > 0,
       validated: ValidatedOptions.default,
       location: '',
     };
   }
 
   private handleCreate(): void {
-    const factory = new FactoryLocationAdapter(this.state.location);
+    const { editorId, editorImage } = this.props;
+    const { location } = this.state;
+
+    const factory = new FactoryLocationAdapter(location);
+
+    // add the editor id and editor image to the URL
+    // if they are not already there
+    if (
+      factory.searchParams.has(EDITOR_ATTR) === false &&
+      factory.searchParams.has(EDITOR_IMAGE_ATTR) === false
+    ) {
+      factory.searchParams.set(EDITOR_ATTR, editorId);
```

Comment:
With these changes, we can't create a new workspace with the custom editor from the ```/.che/che-editor.yaml``` file at all.

Reply:
With these changes, you are still able to create new workspaces with a custom editor. On any dashboard page, paste your repo link into the browser address bar to create a location like this and hit Enter: `https://<che_fqdn>#https://github.com/olexii4/web-java-spring-boot.git`. A new workspace with custom editor will be created within a few minutes.

However, I'm not sure if a `.che/` file in a repository should have precedence over an explicitly defined editor, rather than vice versa.

cc @ibuziuk

Reply:
@akurinnoy ```/.che/che-editor.yaml``` should define the default editor for a Git repo.  Am I right or am I wrong?